woocommerce/woocommerce-ios
# Duplicate Products: Ensure variable properties like description, sale price, and image are copied over to the new product variations

Closes: #8611


## Description

While duplicating variable products, only `attributes` and `regular price` are kept. Other properties like `description`, `image`, or `sale price` are not kept.

This PR ensures that these properties are copied to duplicated product variations.

### Changes

- `Networking/Networking/Model/Product/CreateProductVariation.swift`: Added `salePrice`, `description`, and `images` properties when creating a product variation
- `WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormRemoteActionUseCase.swift`: Added `salePrice`, `description`, and `images` properties to the new product variation before duplicating the product

I also updated some classes that call the `CreateProductVariation` constructor with the additional properties and updated the related unit tests.

## Testing instructions

1. Create a variable product with different variations (descriptions, regular price, sale price, and images). Or you can edit the default variable product descriptions and duplicate the "Hoodie" variable product that comes with the default WooCommerce store.
2. Go to the Product editing screen
3. Tap "..." and choose "Duplicate"
4. Once the app finishes duplicating the product, double-check the properties for product and product variations

Since this PR will also affect variable product creation and generation, please also perform testing in the app that involves the following actions:

1. Create a new variable product
2. Generate product variations based on the attributes added to the product

## Screenshots


### Test duplicating variable product

https://user-images.githubusercontent.com/40906847/228446378-0745c120-80ad-421a-8959-fdb71a1fc71f.mp4

---
